prime-router/metadata/hl7_mapping/OML_O21/datatype/ce-coded-with-exceptions.yml

```diff
@@ -0,0 +1,141 @@
+elements:
+
+  ## This schema supports parsing the code FHIR resource to HL7 two different ways
+  ## 1. Attempt to use the code-index-name to identify the identifier, alternate identifier and second alternate identifier
+  ## 2. Rely on the ordering of the coding array assuming it is [identifier, alternate identifier, second alternate identifier]
+
+  - name: identifier-code
+    condition: >
+      %resource.coding.where(extension(%`rsext-code-index-name`).value = "identifier").exists() or
+      %resource.coding[0].extension(%`rsext-code-index-name`).exists().not()
+    value: [ '%resource.coding.where(extension(%`rsext-code-index-name`).value = "identifier").code', '%resource.coding[0].code' ]
+    hl7Spec: [ '%{cweFieldPath}-1' ]
+  - name: identifier-display
+    condition: >
+      %resource.coding.where(extension(%`rsext-code-index-name`).value = "identifier").exists() or
+      %resource.coding[0].extension(%`rsext-code-index-name`).exists().not()
+    value: [ '%resource.coding.where(extension(%`rsext-code-index-name`).value = "identifier").display', '%resource.coding[0].display' ]
+    hl7Spec: [ '%{cweFieldPath}-2' ]
+  - name: identifier-system-url
+    resource: '%resource.coding.where(extension(%`rsext-code-index-name`).value = "identifier").system'
+    condition: '%resource.value.startsWith("http")'
+    value: [ '%resource.getCodingSystemMapping()' ]
+    hl7Spec: [ '%{cweFieldPath}-3' ]
+  - name: identifier-system-url
+    resource: '%resource.coding[0]'
+    condition: '%resource.system.value.startsWith("http") and %resource.extension(%`rsext-code-index-name`).exists().not()'
+    value: [ '%resource.system.getCodingSystemMapping()' ]
+    hl7Spec: [ '%{cweFieldPath}-3' ]
+  - name: identifier-system-url
+    resource: '%resource.coding[0]'
+    condition: '%resource.system.startsWith("http") and %resource.extension(%`rsext-code-index-name`).exists().not()'
```

Comment:
We were unable to reuse the original code under `common/datatype/ce-coded-with-exception.yml` because the code for `identifier-system-url` breaks the integration test. The `%resource.system` is used by other messages so if the original file is shared, this will produce unwanted behavior such that other messages other than OML, will have their nk1.3 affected. In order to maintain the changes to only the OML message, the changes were made at the OML package level.

Reply:
Just to confirm, this change is because you want to `%resource.system` and not %resource.system.getCodingSystemMapping`?

Could you maybe solve that by adding an entry [here](https://github.com/CDCgov/prime-reportstream/blob/11f98ae573e87eab3ad2dcc286cc4043cb8774e2/prime-router/src/main/kotlin/fhirengine/translation/hl7/utils/CustomFHIRFunctions.kt#L267)?

Reply:
We made changes to the coding mapping method, thanks for the input.